internal/bootstrap/tenant_event_discovery.go

```
@@ -0,0 +1,150 @@
+
+// initTenantEventDiscovery creates the TenantCache, Pub/Sub Redis client, EventDispatcher,
+// and TenantEventListener for event-driven tenant discovery when multi-tenant mode is
+// enabled and a Redis host is configured.
+//
+// Returns:
+//   - listener: the running TenantEventListener (nil if not applicable)
+//   - cache: the shared TenantCache (nil if not applicable)
+//   - cleanup: a function that stops the listener and closes the Pub/Sub client; always non-nil
+//
+// When MULTI_TENANT_ENABLED=false or MULTI_TENANT_REDIS_HOST is empty, this is a no-op.
+func initTenantEventDiscovery(
```

golangci-lint (2.11.4)

[error] calculated cyclomatic complexity for function initTenantEventDiscovery is 14, max is 13

(cyclop)